tests/functional/test_location_change.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import os
+from dbt.tests.util import run_dbt_and_capture
```

Comment:
I think this might be creating an issue with `tests/functional/test_update_column_policy.py` see [test failure](https://github.com/dbt-labs/dbt-bigquery/actions/runs/4048837842/jobs/6966020678#step:7:464). I wonder if we need to set the location back post test?

Reply:
will have that triggered good catch was  had noticed that fresh conversion was having a issue at eod

Reply:
Can we use mock? Pytest has a way to in-runtime override for the test itself. Just checkout other tests in dbt core that use this (if it's db-side may not be possible because that state is of course not sandboxed to test runtime). 

I am concerned a failure mid test operation could fail other tests without us knowing why, that is, unless we can sandbox the environment edit.

Reply:
@VersusFacit any particular test that stood out as a good example to you? happy to play around with it, curious its only the next test that fails

Reply:
I was thinking of something like this `https://github.com/dbt-labs/dbt-snowflake/blob/471d07e8b09bfdbee0ec35a7d69bf85b8dc946ca/tests/unit/test_snowflake_adapter.py#L367`. Should self contain (by mocking) any env changes since they're not "really" happening but you can _mock_ that they are. 

May not work if you want to do an end to end integration test, but I could see attacking this in a manner to avoid malforming the env -- just an idea

Reply:
yeah some of this mocking stuff for our unittest didn't know if it would apply honestly to the pytest will give it a go all for now malforming a profile variable in the directory. actually thought each test formed its own whole little micro project which meant different profiles for each test as well.

Reply:
Well what you said is true on a class level but functions share the class and if you change database state, that doesn't get swapped back either unless you tell it to (although I don't like the idea of changing database state without very a bulletproof rollback safety net in place).

Reply:
also tried using `update_config_file` to change the specific `location` variable mid run which seems closer to the way test originally worked but couldn't get the profiiles.yml to come out correctly.